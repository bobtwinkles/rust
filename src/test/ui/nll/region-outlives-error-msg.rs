// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// This test is explicitly about the quality of error message emitted, not
// detecting a class of error not covered by other tests

#![feature(nll)]
#![feature(in_band_lifetimes)]

struct CookieJar<'data> {
    vec: Vec<String>,
    data: &'data (),
}

impl CookieJar<'data> {
    fn cookie_me(&self, _: &i32) -> Cookie<'data> {
        let next = 22;
        me_want_cookie(next, self)
    }
}

struct Cookie<'jar> {
    jar: &'jar CookieJar<'jar>
}

fn me_want_cookie(next: usize, jar: &'jar CookieJar<'jar>) -> Cookie<'jar> {
    Cookie { jar }
}

fn main() {
    let jar = CookieJar { vec: vec![], data: &() };
    me_want_cookie(0, &jar);
}
