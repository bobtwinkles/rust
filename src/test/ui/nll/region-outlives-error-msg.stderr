error[E0999]: The region for the argument `self` does not outlive the region `'data`. Required due to this expression:
  --> $DIR/region-outlives-error-msg.rs:24:13
   |
LL |         let next = 22;
   |             ^^^^
   |
note: the too-short region is created here
  --> $DIR/region-outlives-error-msg.rs:23:18
   |
LL |     fn cookie_me(&self, _: &i32) -> Cookie<'data> {
   |                  ^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0999`.
